File: hw2/codes/cakin_haci.c
#include <stdio.h>
int main() {
// 1. Iteration statements provided
// Iterations are provided in c by using while, for and do while loops
printf("1. Iteration statements provided \n");
int number = 0;
int trace = 0;
printf("while loop -----\n");
while( trace < 5 ){
    printf("Value of trace : %d in while loop\n", trace);
    trace++;
}
printf("Trace is reached to 5 \n\n");

printf("for loop -----\n");
for(int i = 0; i < 5; i++){
    printf("Value of i : %d in for loop\n", i);
}
printf("i is reached to 5 \n\n");

printf("do while loop -----\n");
do{ 
    printf("Value of number : %d in do-while loop\n", number);
    number++;
}while(number < 5);
printf("number is reached to 5 \n\n\n");
// 2. Data structures suitable for iteration
// Built in data structures are in c array
printf("2. Data structures suitable for iteration \n");
char array1[] = {'h','a','c','i'};
int array2[] = {2,3,5,1,6,3};
for(int i = 0; i < 4; i++){
    printf("%d. letter of haci is %c \n", i+1,array1[i]);
}

printf("\n");
trace = 0;
while(array2[trace] != 6){
    printf("6 is not found until %d. element \n",trace);
    trace++;
}


// 3. The way the next item is accessed
printf("3. The way the next item is accessed");
int num1 = 0;
printf("while---");
while ( num1 < 11 ){
    if( num1 == 6 ){
        printf("num1 Reached to 6\n");
        num1 += 2;
    }
    num1 = num1 + 1;
    printf("Turn %d times", num1);
    printf("\n");
}
printf("\nfor---");
for(int a = 0; a < 10; a = a+2){
    printf("Current value is %d \n",a );
}

    return 0;
}

File: hw2/codes/cakin_haci.rs


fn main() {
// 1. Iteration statements provided
// Iterations are provided in rust by using while, for and loop 
println!("1. Iteration statements provided \n");
let mut num = 0;
while num < 5 {
    if num != 4{
        println!("{} is the current value in while", num);
    }else{
        println!("while reached to 4\n");
        break;
    }
    num = num + 1;
}
let mut num1 = 0;
loop{
    if num1 == 4 {
        println!("loop reached to 4\n");
        break;
    }
    println!("{} is the current value in loop",num1);
    num1 = num1 + 1;
}

for num2 in 0..4{
    println!("{} is the current value in for",num2);
}
println!("for reached to 4");
// 2. Data structures suitable for iteration
// Built in data structures are in rust array
println!("\n2. Data structures suitable for iteration\n");
let array2 = [1907,24,02,2000];
for num4 in array2.iter(){
    println!("Element is: {}", num4 );
}

// 3. The way the next item is accessed
println!("\n3. The way the next item is accessed\n");
let mut num5 = 0;
   while num5 < 11 {
      if num5 == 6 {
        println!("num5 Reached to 6");
        break
    }
    println!("num5 =  {}", num5);
    num5 = num5 + 2;
  }

}
